api/health_check.go

```diff
@@ -15,10 +22,48 @@ type networkInfo struct {
 }
 
 type healthCheckResponse struct {
-	Network networkInfo `json:"network"`
+	CoreIndexer *coreIndexerHealth `json:"core_indexer"`
+	Network     networkInfo        `json:"network"`
+}
+
+type coreIndexerHealth struct {
+	LastIndexedBlock int64 `json:"last_indexed_block"`
+	BlockDiff        int64 `json:"block_diff"`
+}
+
+func (app *ApiServer) getCoreIndexerHealth(ctx context.Context) (*coreIndexerHealth, error) {
+	nodeInfo, err := app.auds.Core.GetNodeInfo(ctx, connect.NewRequest(&corev1.GetNodeInfoRequest{}))
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, "Failed to get core node info")
+	}
+	chainHeight := nodeInfo.Msg.CurrentHeight
+
+	var indexerLastBlockHeight int64
+	err = app.pool.QueryRow(ctx, "SELECT COALESCE(last_checkpoint, 0) FROM indexing_checkpoints WHERE tablename = $1", core_indexer.CoreIndexerCheckpointName).Scan(&indexerLastBlockHeight)
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, "Failed to get core indexer last block height")
+	}
+
+	// Indexer can actually be ahead of the block our RPC returns!
+	blockDiff := int64(math.Max(0, float64(chainHeight-indexerLastBlockHeight)))
```

Comment:
feels kinda like a footgun haha, but i guess... this makes sense?

Reply:
It doesn't really hurt anything if we let it go negative. It was just bugging me :-p
I can remove this and we can just let it be ahead sometimes if it's goin' reeeel fast.